### ISSUE TITLE
longer training time after updating

### ISSUE THREAD

**Issue by YueYin27** [nerfstudio-project/nerfstudio#2127]
Hi, I updated my forked repo last week and found that the training time became around twice longer than before, using the same training script. Besides, the use of GPU memory was also reduced, eval loss increased. Here's my training script:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/0b9de551-b415-46cf-8cad-333fef0bbc19)

Here are the graphs from wandb (pink: before updating; blue: after updating):
![`MKLR3B $H{77%Z4$ _11$I](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/3aecc7bb-9a30-4ac9-82e7-2502f51a54cd)

**Comment by tancik** [comment]
Does your data have masks by any chance?

**Comment by YueYin27** [comment]
> Does your data have masks by any chance?

Hi @tancik , my data doesn't have masks. I also tried the blender-data (lego), but the training time is still longer than before. 

![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/bdf1f015-e3fe-4d94-ac49-6bd19c24ff33)

**Comment by YueYin27** [comment]
Hi @tancik , I just found that when I set `--machine.num-devices` to 1 or 2, the training time was back to normal. However, when I set it to 3 or 4, training time increased accordingly, and so did eval loss. In older version, I set `--machine.num-gpus` to 4, and everything worked fine.

![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/30a43459-a892-4fa6-bec9-d4b59329bc91)

**Comment by pwais** [comment]
1) what's your system RAM like?  approx how many images / megabytes is the dataset?
2) did you change the batch size and/or learning rate when changing the number of GPUs?
3) can you say what model your GPUs are?  they're Pascal arch or newer?

**Comment by YueYin27** [comment]
1. RAM info:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/4b149706-38b3-4f93-9a84-a81aa6244a8a)
200 images in train set, and 200 in val set.
3. No I didn't change anything else except `--machine.num-devices`.
4. Tesla PG500-216, Pascal arch.
![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/bf284dd9-10b1-44b7-bb0d-da6fe39c9de5)

**Comment by pwais** [comment]
Wow I've never heard of that GPU, the web says it's actually Volta not Pascal but I'm not sure.  In any case the Tensor cores on Turing and newer are key for instant-ngp as well as tiny-cuda-nn.  That warning message says arch is 70 which is Volta.  All that said, I don't think `tiny-cuda-nn` defaulting to Cutlass should necessarily impact scaling like you see ...

Looks like you have enough RAM so the issue isn't bottleneck on image I/O.  You have at least say 8 to 16 CPUs right?

This might be a regression in pytorch DDP in the upgrade to 2.x from 1.x ?  While I'm sure V100s get tested a bit, maybe there's something special in your hardware config.   

Lastly have you tried a newer driver / CUDA, like CUDA 11.8 or 12.0 ?  I'm actually not sure pytorch 2.x is supposed to work with CUDA older than 11.7 ?

**Comment by YueYin27** [comment]
> Wow I've never heard of that GPU, the web says it's actually Volta not Pascal but I'm not sure. In any case the Tensor cores on Turing and newer are key for instant-ngp as well as tiny-cuda-nn. That warning message says arch is 70 which is Volta. All that said, I don't think `tiny-cuda-nn` defaulting to Cutlass should necessarily impact scaling like you see ...
> 
> Looks like you have enough RAM so the issue isn't bottleneck on image I/O. You have at least say 8 to 16 CPUs right?
> 
> This might be a regression in pytorch DDP in the upgrade to 2.x from 1.x ? While I'm sure V100s get tested a bit, maybe there's something special in your hardware config.
> 
> Lastly have you tried a newer driver / CUDA, like CUDA 11.8 or 12.0 ? I'm actually not sure pytorch 2.x is supposed to work with CUDA older than 11.7 ?

Yeah I think so, enough RAM.

I haven't tried a newer CUDA. I've talked to people managing the remote server I'm using, they'll build a version with newer packages tomorrow and I'll test if pytorch DDP is causing the issue.

Thank you!

**Comment by machenmusik** [comment]
Not sure if related, 
but I have observed that on the same physical machine, 
using latest nerfstudio with torch 2.0.1+cu118 on Windows (which throws warning about torch.compile)
is actually faster than latest nerfstudio with torch 2.0.1+cu118 on WSL2 Ubuntu (which actually uses torch.compile)
(nerfacto iterations around 20ms vs. 28ms)

**Comment by YueYin27** [comment]
Hi @pwais @machenmusik @tancik , I've tested an environment with newer packages, and nothing changed. I've talked to people who manage the GPU server and he explained the reason! Thank you  @L-Lamont
![image](https://github.com/nerfstudio-project/nerfstudio/assets/75305926/73dbd532-5a9b-445d-bee8-a7e4bb27ad13)

**Comment by YueYin27** [comment]
> I have also encountered this phenomenon. Do you happen to know any methods for reducing training time?

Hi, based on the reasons mentioned above, I just specified 1-2 GPUs to reduce the communication time between CPUs, rather than using 4 GPUs
